client/components/domains/domain-product-price/index.jsx

```diff
@@ -93,8 +102,8 @@ const DomainProductPrice = React.createClass( {
 } );
 
 export default connect(
-	state => ( { domainsWithPlansOnly: getCurrentUser( state ) ?
-		currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY ) :
-		true
+	state => ( { domainsWithPlansOnly: getCurrentUser( state )
+		? currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+		: true
```

Comment:
Funny, it reads pretty clearly to me, but it might be that my brain is broken due to long-term abuse.

The way I intuitively think of `! a || …` is "if there is no `a`, well that's enough [of a condition]; otherwise, we need _something more_. In the case at hand, that means:

`! getCurrentUser( state ) || currentUserHasFlag( state, … )`

which reads:
- no user? that's fine
- otherwise, make sure they have the right flag

This stands in contrast with the common pattern:

`getCurrentUser( state ) && currentUserHasFlag( state, … )`

which reads:
- no user? that's no good, _we needs it_
- also, we need the flags

To put it another way, for me the pattern isn't so much `! a || b` as just `! a ||`. It's a lenient test, or an easy early-bailing mechanism, compared with the strong `a &&` pattern.